pkg/build/scratch.go

```diff
@@ -0,0 +1,168 @@
+/*
+Copyright 2023 Google LLC All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package build
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"hash"
+	"strings"
+
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/partial"
+	"github.com/google/go-containerregistry/pkg/v1/types"
+)
+
+var _ partial.UncompressedImageCore = (*scratchImageContents)(nil)
+var _ partial.WithConfigFile = (*scratchImageContents)(nil)
+var _ v1.ImageIndex = (*scratchImageIndex)(nil)
+
+// ScratchImage returns a scratch image manifest with scratch images for each of the specified platforms
+func ScratchImage(platforms []string) (Result, error) {
```

Comment:
```suggestion
func scratchImage(platforms []string) (Result, error) {
```

I don't think we should export this method so that external projects could use it.

Reply:
I'd need to move it to the commands package if its not exported. WDYT?

Reply:
That's fine. If we need it in more than one package, we can move it to `pkg/internal/something`.